src/main/drivers/transponder_ir.c

```diff
@@ -39,6 +44,135 @@ uint8_t transponderIrDMABuffer[TRANSPONDER_DMA_BUFFER_SIZE];
 
 volatile uint8_t transponderIrDataTransferInProgress = 0;
 
+static IO_t transponderIO = IO_NONE;
+static TIM_TypeDef *timer = NULL;
+#if defined(STM32F3)
+static DMA_Channel_TypeDef *dmaChannel = NULL;
+#elif defined(STM32F4)
+static DMA_Stream_TypeDef *stream = NULL;
+#else
+#error "Transponder not supported on this MCU."
+#endif
+
+static void TRANSPONDER_DMA_IRQHandler(dmaChannelDescriptor_t* descriptor)
+{
+    if (DMA_GET_FLAG_STATUS(descriptor, DMA_IT_TCIF)) {
+        transponderIrDataTransferInProgress = 0;
+#if defined(STM32F3)
+        DMA_Cmd(descriptor->channel, DISABLE);
+#elif defined(STM32F4)
+        DMA_Cmd(descriptor->stream, DISABLE);
+#endif
+        DMA_CLEAR_FLAG(descriptor, DMA_IT_TCIF);
+    }
+}
+
+void transponderIrHardwareInit(ioTag_t ioTag)
+{
+    if (!ioTag) {
+        return;
+    }
+
+    TIM_TimeBaseInitTypeDef  TIM_TimeBaseStructure;
+    TIM_OCInitTypeDef  TIM_OCInitStructure;
+    DMA_InitTypeDef DMA_InitStructure;
+
+    const timerHardware_t *timerHardware = timerGetByTag(ioTag, TIM_USE_ANY);
+    timer = timerHardware->tim;
+
+#if defined(STM32F3)
+    if (timerHardware->dmaChannel == NULL) {
+        return;
+    }
+#elif defined(STM32F4)
+    if (timerHardware->dmaStream == NULL) {
+        return;
+    }
+#endif
+
+    transponderIO = IOGetByTag(ioTag);
+    IOInit(transponderIO, OWNER_TRANSPONDER, 0);
+    IOConfigGPIOAF(transponderIO, IO_CONFIG(GPIO_Mode_AF, GPIO_Speed_50MHz, GPIO_OType_PP, GPIO_PuPd_DOWN), timerHardware->alternateFunction);
+
+    dmaInit(timerHardware->dmaIrqHandler, OWNER_TRANSPONDER, 0);
+    dmaSetHandler(timerHardware->dmaIrqHandler, TRANSPONDER_DMA_IRQHandler, NVIC_PRIO_TRANSPONDER_DMA, 0);
+
+    RCC_ClockCmd(timerRCC(timer), ENABLE);
+
+    /* Time base configuration */
+    TIM_TimeBaseStructInit(&TIM_TimeBaseStructure);
+    TIM_TimeBaseStructure.TIM_Period = 156;
```

Comment:
@hydra: I think there is interest from users with F4 boards in using IR transponders, and thanks to resource management, they will be able to use any free pin to drive them once the code gets added to targets, so we should consider adding it.